django/views/defaults.py

```diff
@@ -11,6 +11,7 @@
 ERROR_403_TEMPLATE_NAME = '403.html'
 ERROR_400_TEMPLATE_NAME = '400.html'
 ERROR_500_TEMPLATE_NAME = '500.html'
+HTML_DOCUMENT = '<!DOCTYPE html><html lang="en"><body>{}</body></html>'
```

Comment:
`<head>` and `<title>` are missing if we want to have a valid HTML5 page, maybe
```python
ERROR_PAGE_TEMPLATE = """
<!DOCTYPE html>
<html lang="en">
<head>
  <title>%(title)s</title>
</head>
<body>
  <h1>%(title)s</h1><p>%(details)s</p>
</body></html>
"""
```

Reply:
I'd also go for `<!doctype html>` (lowercased).

Reply:
> I'd also go for `<!doctype html>` (lowercased).

Uppercase is used in other templates in Django.

Reply:
> I'd also go for <!doctype html> (lowercased).

While the doctype is case insensitive, all examples in the [WHATWG spec](https://html.spec.whatwg.org/#the-doctype) spell this as: `<!DOCTYPE html>`.  When given a choice like this, I think we should follow community conventions and spec examples. One benefit, by reducing unnecessary differences, we leak fewer implementation detail that the web application is backed by a specific technology.

Is there any reason to explicitly prefer lowercase?

Reply:
Not really, but it is not our first lowercased doctype (see [default_urlconf.html](https://github.com/django/django/blob/master/django/views/templates/default_urlconf.html)).